app/models/assignment.rb

```diff
@@ -1464,4 +1464,68 @@ def same_membership_as_csv_row?(row, existing_grouping)
     end
   end
 
+  # Reformatted logic for download_assignment_list controller method
+  # Returns an output file for controller to handle.
+  def self.get_assignment_list(file_format, default_fields)
+    assignments = self.all
+
+    case file_format
+    when 'yml'
+      map = {}
+      map[:assignments] = []
+      assignments.map do |assignment|
+        m = {}
+        default_fields.length.times do |i|
+          m[default_fields[i]] = assignment.send(default_fields[i])
+        end
+        map[:assignments] << m
+      end
+      output = map.to_yaml
+    when 'csv'
+      output = MarkusCSV.generate(assignments) do |assignment|
+        default_fields.map do |f|
+          assignment.send(f.to_s)
+        end
+      end
+    end
+    return output
+  end
+
+  def self.upload_assignment_list(file_format, assignment_list, default_fields)
+
```

Comment:
Remove this blank line. Also, the indentation below is off (should be indented inside the `def`).